--- spice-gtk-0.27/spice-common/common/generated_client_demarshallers.c	2014-12-11 08:43:53.000000000 -0500
+++ spice-gtk/spice-common/common/generated_client_demarshallers.c	2015-02-03 20:44:50.484964256 -0500
@@ -64,10 +64,10 @@
 #define write_int32(ptr, val) (*((int32_t *)(ptr))) = val
 #define read_uint32(ptr) (*((uint32_t *)(ptr)))
 #define write_uint32(ptr, val) (*((uint32_t *)(ptr))) = val
-#define read_int64(ptr) (*((int64_t *)(ptr)))
-#define write_int64(ptr, val) (*((int64_t *)(ptr))) = val
-#define read_uint64(ptr) (*((uint64_t *)(ptr)))
-#define write_uint64(ptr, val) (*((uint64_t *)(ptr))) = val
+static inline int64_t read_int64(uint8_t* ptr) { int64_t val; memcpy(&val,ptr,sizeof(int64_t)); return val;}
+#define write_int64(ptr,v) { int64_t val = v; memcpy(ptr, &val, sizeof(int64_t)); }
+static inline uint64_t read_uint64(uint8_t* ptr) { uint64_t val; memcpy(&val,ptr,sizeof(uint64_t)); return val;}
+#define write_uint64(ptr,v) { uint64_t val = v; memcpy(ptr, &val, sizeof(uint64_t)); }
 #endif
 
 static int8_t SPICE_GNUC_UNUSED consume_int8(uint8_t **ptr)
--- spice-gtk-0.27/spice-common/common/marshaller.c	2014-12-11 08:36:22.000000000 -0500
+++ spice-gtk/spice-common/common/marshaller.c	2015-02-03 20:44:24.300370070 -0500
@@ -41,8 +41,14 @@
 #define write_uint16(ptr,v) (*((uint16_t *)(ptr)) = v)
 #define write_int32(ptr,v) (*((int32_t *)(ptr)) = v)
 #define write_uint32(ptr,v) (*((uint32_t *)(ptr)) = v)
-#define write_int64(ptr,v) (*((int64_t *)(ptr)) = v)
-#define write_uint64(ptr,v) (*((uint64_t *)(ptr)) = v)
+/*
+ * Thanks to shohyanglim@gmail.com for saving me time and discovering these
+ * hacks to avoid SIGBUS on some ARM processors.
+ *
+ * TODO: Find out whether something better can be done to avoid the SIGBUS.
+ */
+#define write_int64(ptr,v) { int64_t val = v; memcpy(ptr, &val, sizeof(int64_t)); }
+#define write_uint64(ptr,v) { uint64_t val = v; memcpy(ptr, &val, sizeof(uint64_t)); }
 #endif
 
 typedef struct {
